test/rockpaperscissors.js

```diff
@@ -0,0 +1,717 @@
+const truffleAssert = require('truffle-assertions');
+const timeMachine = require('ganache-time-traveler');
+const Game = artifacts.require("./RockPaperScissors.sol");
+
+contract('rcp', async accounts => {
+
+    const { toBN, soliditySha3 } = web3.utils;
+    const { getBalance, getTransaction } = web3.eth;
+    const playerOneSecretString = "secret1";
+    const playerTwoSecretString = "secret2";
+    const wrongSecretString = "wrongSecret";
+    
+    const ROCK = 0;
+    const PAPER = toBN(1);
+    const SCISSORS = toBN(2);
+
+    const INIT = "0";
+    const PLAYER_ONE_MOVE = "1";
+    const PLAYER_TWO_MOVE = "2";
+    const PLAYER_ONE_REVEAL = "3";
+
+    const INVALID_MOVE_MSG = "Invalid move";
+    const INVALID_STEP_MSG = "Invalid step";
+    const HASH_MISMATCH_MSG = "Move and secret do not match";
+    const GAME_NOT_EXPIRED_MSG = "Game has not expired";
+    const MOVE_HASH_EMPTY_MSG = "Move hash is empty";
+    const SECRET_EMPTY_MSG = "Secret is empty";
+    const INVALID_PLAYER_MSG = "Invalid player";
+
+    const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+    const ZERO_BYTES_32 = "0x0000000000000000000000000000000000000000000000000000000000000000";
+    const DAY_24_HOUR_IN_SECS = 86400;
+
+    const getBlockTimeStamp = async(txObj) => {
+
+        const tx = await getTransaction(txObj.tx);
+        const blockData = await web3.eth.getBlock(tx.blockNumber)
+
+        return blockData.timestamp;
+    }
+
+    const checkEventNotEmitted = async () => {
+        const result = await truffleAssert.createTransactionResult(rcp, rcp.transactionHash);
```

Comment:
This checks that there were no events emitted during the creation of `rcp`.
However, you peppered this function everywhere. So it is not doing what you ostensibly expect it to be doing.

Reply:
Yes. You are right. I need to investigate further.

Reply:
I have removed unnecessary use of `truffleAssert.eventNotEmitted()` and added it where it adds value.